macro/src/lib.rs

```diff
@@ -232,9 +232,18 @@ fn fetch_metadata(args: &RuntimeMetadataArgs) -> Result<subxt_codegen::Metadata,
                 )
             }
 
-            let root = std::env::var("CARGO_MANIFEST_DIR").unwrap_or_else(|_| ".".into());
-            let root_path = std::path::Path::new(&root);
-            let path = root_path.join(rest_of_path);
+            // Replace $OUT_DIR with the actual OUT_DIR environment variable
+            let path = if rest_of_path.contains("$OUT_DIR") {
+                let out_dir = std::env::var("OUT_DIR").unwrap_or_else(|_| {
+                    abort_call_site!("$OUT_DIR is used in runtime_metadata_path but OUT_DIR environment variable is not set")
+                });
+                std::path::Path::new(&rest_of_path.replace("$OUT_DIR", &out_dir)).into()
+            } else {
+                let root = std::env::var("CARGO_MANIFEST_DIR").unwrap_or_else(|_| ".".into());
+                let root_path = std::path::Path::new(&root);
+                root_path.join(rest_of_path)
+            };
```

Comment:
We should probably make a little function for this and apply the same logic above for lines 216-218; people would expect to be able to do the same thing for the runtime-wasm-path field